aws-rds-eventsubscription/src/main/java/software/amazon/rds/eventsubscription/UpdateHandler.java

```diff
@@ -11,44 +12,58 @@
-                sourceId -> proxy.initiate("rds::add-source-id-event-subscription", proxyClient, model, callbackContext)
-                    .translateToServiceRequest((resourceModel) -> Translator.addSourceIdentifierToSubscriptionRequest(resourceModel, sourceId))
-                    .makeServiceCall((addSourceIdentifierToSubscriptionRequest, proxyCall) -> proxyCall.injectCredentialsAndInvokeV2(addSourceIdentifierToSubscriptionRequest, proxyCall.client()::addSourceIdentifierToSubscription))
-            );
-            return progress;
-          })
-          .then(progress -> {
-            Sets.difference(previousSourceIds, currentSourceIds).forEach(
-                sourceId -> proxy.initiate("rds::remove-source-id-event-subscription", proxyClient, model, callbackContext)
-                    .translateToServiceRequest((resourceModel) -> Translator.removeSourceIdentifierFromSubscriptionRequest(resourceModel, sourceId))
-                    .makeServiceCall((removeSourceIdentifierFromSubscriptionRequest, proxyCall) -> proxyCall.injectCredentialsAndInvokeV2(removeSourceIdentifierFromSubscriptionRequest, proxyCall.client()::removeSourceIdentifierFromSubscription))
-            );
-            return progress;
-          })
-          .then(progress -> waitForEventSubscription(proxy, proxyClient, progress))
-          .then(progress -> tagResource(proxy, proxyClient, progress, mergeMaps(request.getSystemTags(), request.getDesiredResourceTags())))
-          .then(progress -> new ReadHandler().handleRequest(proxy, request, callbackContext, proxyClient, logger));
+        final Map<String, String> previousTags = Tagging.mergeTags(
+                request.getPreviousSystemTags(),
+                request.getPreviousResourceTags()
+        );
+        final Map<String, String> desiredTags = Tagging.mergeTags(
+                request.getSystemTags(),
+                request.getDesiredResourceTags()
+        );
+
+        return proxy.initiate("rds::update-event-subscription", proxyClient, model, callbackContext)
+                .translateToServiceRequest(Translator::modifyEventSubscriptionRequest)
+                .makeServiceCall((modifyEventSubscriptionRequest, proxyInvocation) -> proxyInvocation.injectCredentialsAndInvokeV2(modifyEventSubscriptionRequest, proxyInvocation.client()::modifyEventSubscription))
+                .stabilize((modifyEventSubscriptionRequest, modifyEventSubscriptionResponse, proxyInvocation, resourceModel, context) ->
+                        isStabilized(resourceModel, proxyInvocation))
+                .handleError((deleteRequest, exception, client, resourceModel, ctx) -> Commons.handleException(
+                        ProgressEvent.progress(resourceModel, ctx),
+                        exception,
+                        DEFAULT_EVENT_SUBSCRIPTION_ERROR_RULE_SET))
+                .progress()
+                .then(progress -> {
+                    Sets.difference(currentSourceIds, previousSourceIds).forEach(
```

Comment:
nit: this is a good candidate for a well-named variable instead of a direct `Sets.difference` invocation: one needs to read the entire block to figure out that we're adding new tags. Same for the block below. An alternative would be to declare 2 local variables before we perform these invocations.